cmake error on ubuntu16.04

When I used cmake to compile in ubuntu16.04, the following error occurred：

~/libcmaes/build$ cmake .. -DCMAKE_INSTALL_PREFIX=~/.local/

-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Found Eigen3: /usr/include/eigen3 (Required is at least version "3.0.0") 
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found version "2.7.12") 
CMake Error at CMakeLists.txt:82 (target_include_directories):
  target_include_directories called with non-compilable target type


-- Found NumPy: /usr/lib/python2.7/dist-packages/numpy/core/include (found version "1.11.0") 
CMake Error at cmake/FindNumPy.cmake:70 (target_include_directories):
  Cannot specify include directories for imported target "Python::NumPy".
Call Stack (most recent call first):
  CMakeLists.txt:87 (find_package)


-- Looking for component python-py27
CMake Error at CMakeLists.txt:138 (target_include_directories):
  target_include_directories called with non-compilable target type


-- Found boost-python: /usr/lib/x86_64-linux-gnu/libboost_python-py27.so
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring incomplete, errors occurred!
See also "/home/ycj/libcmaes/build/CMakeFiles/CMakeOutput.log".
See also "/home/ycj/libcmaes/build/CMakeFiles/CMakeError.log".

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,8 +15,8 @@ endif ()
 
 # ---------- options ----------
 option (LIBCMAES_BUILD_SHARED_LIBS "Build libcmaes as a shared library" ON)
-option (LIBCMAES_BUILD_PYTHON "build python bindings" ${LIBCMAES_TOP_LEVEL})
-option (LIBCMAES_BUILD_TESTS "build tests" ${LIBCMAES_TOP_LEVEL})
+option (LIBCMAES_BUILD_PYTHON "build python bindings" OFF)
+option (LIBCMAES_BUILD_TESTS "build tests" OFF)
 option (LIBCMAES_BUILD_EXAMPLES "build examples" ${LIBCMAES_TOP_LEVEL})
 option (LIBCMAES_USE_OPENMP "Use OpenMP for multithreading" ON)
 option (LIBCMAES_ENABLE_SURROG "support for surrogates" ON)
@@ -78,10 +78,11 @@ if (LIBCMAES_BUILD_PYTHON)
   find_package (PythonInterp)
   find_package (PythonLibs)
   if (NOT TARGET Python::Module AND PYTHONLIBS_FOUND)
-    add_library (Python::Module UNKNOWN IMPORTED)
-    target_include_directories (Python::Module INTERFACE ${PYTHON_INCLUDE_DIRS})
-    set_target_properties (Python::Module PROPERTIES IMPORTED_LOCATION
-                                                     ${PYTHON_LIBRARIES})
+    add_library (Python::Module SHARED IMPORTED)
+    set_target_properties (
+      Python::Module
+      PROPERTIES IMPORTED_LOCATION ${PYTHON_LIBRARIES}
+                 INTERFACE_INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_DIRS})
   endif ()
 
   find_package (NumPy)
@@ -122,22 +123,15 @@ if 'real_prefix' in dir(sys):
   endif ()
 
   # boost-python
-  set (BOOST_PYTHON_COMPONENT_NAMES
-       python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}
-       python${PYTHON_VERSION_MAJOR} python)
-  foreach (NAME ${BOOST_PYTHON_COMPONENT_NAMES})
-    if (NOT Boost_FOUND)
-      message (STATUS "Looking for component ${NAME}")
-      find_package (Boost QUIET COMPONENTS ${NAME})
-    endif ()
-  endforeach ()
+  FIND_PACKAGE(Boost COMPONENTS python)
   if (Boost_FOUND)
     set (HAVE_BOOST_PYTHON TRUE)
     if (NOT TARGET Boost::python)
-      add_library (Boost::python UNKNOWN IMPORTED)
-      target_include_directories (Boost::python INTERFACE ${Boost_INCLUDE_DIRS})
-      set_target_properties (Python::Module PROPERTIES IMPORTED_LOCATION
-                                                       ${Boost_LIBRARIES})
+      add_library (Boost::python SHARED IMPORTED)
+      set_target_properties (
+        Boost::python
+        PROPERTIES IMPORTED_LOCATION ${Boost_LIBRARIES}
+                   INTERFACE_INCLUDE_DIRECTORIES ${Boost_INCLUDE_DIRS})
     endif ()
     message (STATUS "Found boost-python: ${Boost_LIBRARIES}")
   endif ()

--- a/README.md
+++ b/README.md
@@ -61,7 +61,7 @@ For compiling with CMake:
 ```
 mkdir build
 cd build
-cmake .. -DCMAKE_INSTALL_PREFIX=~/.local/ .
+cmake .. -DCMAKE_INSTALL_PREFIX=~/.local/
 make -j2
 make install
 ```
@@ -141,6 +141,12 @@ sudo apt-get install libboost-python-dev
 make
 make install
 ```
+
+and with `cmake`:
+```
+cmake .. -DCMAKE_INSTALL_PREFIX=~/.local/ -DLIBCMAES_BUILD_PYTHON=ON
+```
+
 - test the bindings:
 ```
 cd python

--- a/cmake/FindNumPy.cmake
+++ b/cmake/FindNumPy.cmake
@@ -67,5 +67,6 @@ find_package_handle_standard_args (
 
 if (NumPy_FOUND AND NOT TARGET Python::NumPy)
   add_library (Python::NumPy INTERFACE IMPORTED)
-  target_include_directories (Python::NumPy INTERFACE ${NumPy_INCLUDE_DIRS})
+  set_target_properties (Python::NumPy PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+                                                  ${NumPy_INCLUDE_DIRS})
 endif ()

